### PR TITLE
Add anonymize endpoint to return an anonymized content of a pdf file sent in post req

### PR DESCRIPTION
**Pull request by romaissaMe**
@achab

### CODE REVIEW

**Comment by achab** [comment]
Thanks for the PR @romaissaMe ! Can you add to the README a command to run the API?

File: app/api/main.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from fastapi import FastAPI
+from routers import anonymize_file
```
Comment (achab) [inline app/api/main.py:2]: it's more "pythonic" to import like that

```python
from routers.anonymize_file import router
```

Comment (romaissaMe) [reply app/api/main.py:2]: it was done that way cuz we may have several routers so can't assign for each router a unique name thus kept the parent file , tell me if you still wanna me change it @achab

Comment (achab) [reply app/api/main.py:2]: you can also rename `router` to `anonymize_router` in the file `app/api/routers/anonymize_file.py`. when possible, it's better to import using full path

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 anonymized_text.txt
 __pycache__
 .pytest_cache
+.vscode
```
Comment (achab) [inline .gitignore:5]: :+1:

File: requirements.txt [hunk 3]
```diff
@@ -5,3 +5,6 @@ en_core_web_lg==3.5.0
 pdfplumber==0.9.0
 pytest==7.3.2
 openai==0.27.8
+python-multipart==0.0.6
```
Comment (achab) [inline requirements.txt:8]: where do you use this library?

Comment (romaissaMe) [reply requirements.txt:8]: when I was searching how to do that task with fastAPI it was required to be installed to handle multipart/form-data request but yes it was not used!

File: app/api/main.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from fastapi import FastAPI
+from routers.anonymize_file import anonymize_router
```
Comment (achab) [inline app/api/main.py:2]: replace `from routers` with `from app.api.routers`

File: README.md [hunk 5]
```diff
@@ -2,8 +2,10 @@
...
-Helper functionsfor converting pdf file to txt using pdfplumber and reading from txt files  
+Helper functions for converting pdf file to txt using pdfplumber and reading from txt files  
 ## anonymize_test
 Running unit tests on anonymize.py using pytest
 ## installing requirements using pip
-pip install -r requirements.txt
\ No newline at end of file
+pip install -r requirements.txt
+## To run the API
+uvicorn main:app --reload
```
Comment (achab) [inline README.md:11]: to be replaced with `uvicorn app.api.main:app --reload`

**Comment by romaissaMe** [comment]
@achab

**Review by achab — COMMENTED** [review]
Very good PR @romaissaMe ! Before merging, could you add a command to use the endpoint? Something like:

- Once the API is running, run the following command to anonymize a file.
```bash
curl http://localhost:8000/anonymize -F "file=@tests/example.pdf"
```

File: app/api/routers/anonymize_file.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import os
+import shutil
+from fastapi import APIRouter, File, Form, UploadFile
+from app.utils.anonymize import anonymize
+
+
+anonymize_router = APIRouter()
+
+@anonymize_router.post("/anonymize")
+async def anonymize_file(name:str=Form(...),file:UploadFile=File(...)):
```
Comment (achab) [inline app/api/routers/anonymize_file.py:10]: parameter `name` to be removed here

File: app/api/routers/anonymize_file.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import os
+import shutil
+from fastapi import APIRouter, File, Form, UploadFile
```
Comment (achab) [inline app/api/routers/anonymize_file.py:3]: `Form` to be removed, since `name` parameter is useless

**Review by achab — APPROVED** [review]
Great work!

**Pull request merged**